endpoint-common/src/auth.rs

```diff
@@ -72,17 +73,21 @@ impl DeviceAuthenticator {
         tenant: T,
         device: D,
         credential: Credential,
+        service_auth_token: String,
```

Comment:
Wouldn't it be possible to add the token to the client itself? Instead of adding it to each function call.

Reply:
I thought of it and i found it werid to have a service token into the `DeviceAuthenticator` instance. But maybe i am odd for finding that odd :D